vacationtime-solution/src/Code.js

```diff
@@ -0,0 +1,195 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var COLUMN_NUMBER = {
+  EMAIL: 2,
+  NAME: 3,
+  MANAGER_EMAIL: 4,
+  START_DATE: 5,
+  END_DATE: 6,
+  APPROVAL: 8,
+  NOTIFIED: 9,
+};
+
+var REJECTION_EMAIL_MESSAGE = 'Your vacation time request was not approved.';
+var REJECTION_EMAIL_SUBJECT = 'ERR: Vacation Time Request NOT Approved';
+
+/**
+ * Add custom menu items when opening the sheet.
+ */
+function onOpen() {
+  var sheetUi = SpreadsheetApp.getUi();
+  sheetUi.createMenu('TimeOff')
+      .addItem('Manager Approval', 'managerApproval')
+      .addItem('Notify Employees', 'notifyEmployees')
+      .addToUi();
+}
+
+/**
+ * Adds manager approval column with drop-down options.
+ */
+function managerApproval() {
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastCol = sheet.getLastColumn();
+  var lastRow = sheet.getLastRow();
+  var frozenRows = sheet.getFrozenRows();
+  var startRow = frozenRows + 1;
+  var numRows = lastRow - frozenRows;
+
+  // Creates approval column.
+  sheet.insertColumnAfter(lastCol);
+  sheet.getRange(frozenRows, COLUMN_NUMBER.APPROVAL)
+      .setValue('APPROVAL');
+
+  // Sets drop-down menu cells in approval column.
+  var approvalColumnRange = sheet.getRange(startRow, COLUMN_NUMBER.APPROVAL,
+      numRows, 1);
+  var dropdownValues = ['APPROVED', 'NOT APPROVED', 'IN PROGRESS'];
+  var rule = SpreadsheetApp.newDataValidation().requireValueInList(dropdownValues)
+      .build();
+  approvalColumnRange.setDataValidation(rule);
+  approvalColumnRange.setValue('IN PROGRESS');
+}
+
+/** 
+ * Creates Notified Column.
```

Comment:
This is the same information provided by the method name, let's give a bit more description on what's going on inside the function.